packages/next/server/next-dev-server.js

```diff
@@ -195,6 +201,28 @@ export default class DevServer extends Server {
     if (finished) {
       return
     }
+    const { pathname } = parsedUrl
+
+    // check for a public file, throwing error if there's a
+    // conflicting page
+    if (this.nextConfig.experimental.publicDirectory) {
+      if (await this.hasPublicFile(pathname)) {
```

Comment:
Shouldn't this be added before `this.hotReloader.run` above 🤔

Reply:
Updated with what we discussed 👍

Reply:
This should still be moved up

Reply:
Moved